OpenRA.Game/Sync.cs

```diff
@@ -161,20 +168,30 @@ public static int HashUsingHashCode<T>(T t)
 			return t.GetHashCode();
 		}
 
+		public static void RunUnsynced(World world, Action fn)
+		{
+			RunUnsynced(debugSettings.SyncCheckUnsyncedCode, world, () => { fn(); return true; });
```

Comment:
Why can't this be
```suggestion
			RunUnsynced(debugSettings.SyncCheckUnsyncedCode, world, () => fn());
```
?

Reply:
Since you want to call the Func<bool> variant, not itself recursively.

Reply:
Yeah, but don't `() => { fn(); return true; }` and `() => fn()` both call the action once?

Reply:
The first calls the Func<bool> variant.

Reply:
Isn't this exactly the kind of thing that the compiler or JIT should be optimizing away? Are you sure that this actually makes a real world difference?

Reply:
Since we can't change the second case either (I had overlooked that), I wouldn't mind having it here either. I just found the construct odd.

Reply:
This change is mostly about not having 20 places in the code where the same argument is passed. It doesn't do much good performance wise.

Regarding the Action and Func. I had hoped the dotnet could do auto conversion from an Action to Func<void> but that is not the case. So the extra call is needed. But let's hope yes that the compiler/JIT will detect that.

Reply:
https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-6/ .NET6 makes some significant improvements.